examples/tv-casting-app/darwin/MatterTvCastingBridge/MatterTvCastingBridge/MCIdentificationDeclarationOptions.mm

```diff
@@ -0,0 +1,179 @@
+/**
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#import "MCIdentificationDeclarationOptions.h"
+
+#import "core/IdentificationDeclarationOptions.h" // from tv-casting-common
+#import "core/Types.h" // from tv-casting-common
+#import "include/CHIPProjectAppConfig.h" // from tv-casting-common
+
+@interface MCIdentificationDeclarationOptions ()
+
+// Private properties:
+
+/**
+ * Feature: Target Content Application - Flag to instruct the Commissioner not to display a
+ * Passcode input dialog, and instead send a CommissionerDeclaration message if a commissioning
+ * Passcode is needed.
+ */
+@property (nonatomic) BOOL noPasscode;
+/**
+ * Feature: Coordinate Passcode Dialogs - Flag to instruct the Commissioner to send a
+ * CommissionerDeclaration message when the Passcode input dialog on the Commissioner has been
+ * shown to the user.
+ */
+@property (nonatomic) BOOL cdUponPasscodeDialog;
+/**
+ * Feature: Commissioner-Generated Passcode - Flag to instruct the Commissioner to use the
+ * Commissioner-generated Passcode for commissioning.
+ */
+@property (nonatomic) BOOL commissionerPasscode;
+/**
+ * Feature: Commissioner-Generated Passcode - Flag to indicate whether or not the Commissionee has
+ * obtained the Commissioner Passcode from the user and is therefore ready for commissioning.
+ */
+@property (nonatomic) BOOL commissionerPasscodeReady;
+/**
+ * Feature: Coordinate Passcode Dialogs Flag - to indicate when the Commissionee user has decided
+ * to exit the commissioning process.
+ */
+@property (nonatomic) BOOL cancelPasscode;
+/**
+ * Feature: Target Content Application - The set of content app Vendor IDs (and optionally,
+ * Product IDs) that can be used for authentication. Also, if TargetAppInfo is passed in,
+ * VerifyOrEstablishConnection() will force User Directed Commissioning, in case the desired
+ * TargetApp is not found in the on-device CastingStore.
+ */
+@property (nonatomic, strong) NSMutableArray<MCTargetAppInfo *> * targetAppInfos;
```

Comment:
You have a comment at the beginning saying "// Private properties", but these are all public. To actually make them "private" or internal-only, move them to the Internal header file.

Reply:
Something like this "identificationDeclarationOptions.commissionerPasscode = false" will lead to a compile error: Value of type 'MCIdentificationDeclarationOptions' has no member 'commissionerPasscode'